src/lib/albertcore/src/albert/mainwindow/mainwindow.cpp

```diff
@@ -227,6 +227,12 @@ void MainWindow::setVisible(bool visible) {
 /** ***************************************************************************/
 void MainWindow::toggleVisibility() {
    setVisible(!isVisible());
+   // albert might fail to grab keyboard focus if you press keyboard immediately right after HOTKEY was triggered.
+   QTimer::singleShot(1, this, &MainWindow::regrabFocus);
```

Comment:
How did you track this one down? What happens exactly?

Reply:
I'd like to know if this really is an issue or just a timing problem. Could you please check if this is a problem for you if you disable all the extensions?

Reply:
You r right, I turned off all plugins, and it seems to be working fine...
Then I enabled the plugins one by one, problem appeared when External extensions were enabled

Reply:
Yes they are tremendously slow. Thats the reason why setup and teardown will be dropped for extexts.